rust/src/serialization.rs

```diff
@@ -3698,6 +3698,63 @@ impl Deserialize for NetworkId {
     }
 }
 
+impl cbor_event::se::Serialize for Datum {
```

Comment:
@rooooooooob I'm not sure what was up with this `TxOutputData` type. The `datum_option` type is part of the binary spec so I just replaced it with this and manually added the (de)serialization for it